File: OBJC/SFAttributedStringDemo/ViewController.m
//
//  ViewController.m
//  SFAttributedStringDemo
//
//  Created by MeterWhite on 2020/7/17.
//  Copyright © 2020 Meterwhite. All rights reserved.
//

#import "ViewController.h"
#import "SFAttributedString.h"

@interface ViewController ()
@property (unsafe_unretained, nonatomic) IBOutlet UILabel *lb;
@property (unsafe_unretained, nonatomic) IBOutlet UIButton *bt;
@property (unsafe_unretained, nonatomic) IBOutlet UITextView *txtv;
@property (unsafe_unretained, nonatomic) IBOutlet UITextField *txtfd;
@property (unsafe_unretained, nonatomic) IBOutlet UILabel *lb2;

@end

@implementation ViewController

- (instancetype)initWithCoder:(NSCoder *)coder
{
    [self configBeforeUse];
    self = [super initWithCoder:coder];
    return self;
}

- (void)configBeforeUse {
    [SFAtStringCore registerAttributes:@{
        NSForegroundColorAttributeName : UIColor.systemBlueColor,
        NSFontAttributeName : [UIFont systemFontOfSize:21 weight:(UIFontWeightMedium)],
        NSUnderlineStyleAttributeName : @(NSUnderlineStyleSingle)
    } forLabel:@"A"];
    [SFAtStringCore registerAttributes:@{
        NSForegroundColorAttributeName : UIColor.darkGrayColor,
        NSFontAttributeName : [UIFont systemFontOfSize:16 weight:(UIFontWeightRegular)],
    } forLabel:@"B"];
    [SFAtStringCore registerAttributes:@{
        NSForegroundColorAttributeName : UIColor.systemRedColor,
        NSFontAttributeName : [UIFont systemFontOfSize:21 weight:(UIFontWeightMedium)],
        NSUnderlineStyleAttributeName : @(NSUnderlineStyleSingle)
    } forLabel:@"A1"];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    NSString *string_sf = @"[A]Give [[!]star] to [B]SFAttributedString";
//    NSString *string_sf = @"[A][[!]star]012345[[!]star,0,-5.00,21,21]6789[B]][[][[!]][[[[!]star,0,-5.00,21,21]";
    NSLog(@"%@",string_sf.sf_unformattedString);
    [_lb2 setSf_text:string_sf];
    [_lb2 setSf_text:string_sf];
}


@end

File: SWIFT/SFAttributedStringSwiftDemo/ViewController.swift
//
//  ViewController.swift
//  SFAttributedStringSwiftDemo
//
//  Created by MeterWhite on 2020/7/18.
//  Copyright © 2020 Meterwhite. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var lb: UILabel!
    
    @IBOutlet weak var bt: UIButton!
    
    @IBOutlet weak var txtv: UITextView!
    
    @IBOutlet weak var txtfd: UITextField!
    
    @IBOutlet weak var lb2: UILabel!
    
    required init?(coder: NSCoder) {
        ViewController.configBeforeUse()
        super.init(coder: coder)
    }
    
    static func configBeforeUse() {
        SFAtStringCore.registerAttributes([
            NSAttributedString.Key.foregroundColor : UIColor.systemBlue,
            NSAttributedString.Key.font : UIFont.systemFont(ofSize: 21, weight: .medium),
            NSAttributedString.Key.underlineStyle:NSUnderlineStyle.single.rawValue
            ], forLabel: "A")
        SFAtStringCore.registerAttributes([
            NSAttributedString.Key.foregroundColor : UIColor.darkGray,
            NSAttributedString.Key.font : UIFont.systemFont(ofSize: 16, weight: .regular),
            ], forLabel: "B")
        SFAtStringCore.registerAttributes([
        NSAttributedString.Key.foregroundColor : UIColor.systemRed,
        NSAttributedString.Key.font : UIFont.systemFont(ofSize: 21, weight: .medium),
        NSAttributedString.Key.underlineStyle:NSUnderlineStyle.single.rawValue
        ], forLabel: "A1")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let sf = "[A][[!]star]012345[[!]star,0,-5.00,21,21]6789[B]][[][[!]][[[[!]star,0,-5.00,21,21]"
        print(sf.sf_unformattedString as Any)
        lb2.sf_text = sf
        lb2.sf_text = sf
    }
}

